Fix CMake warnings

zann1x opened the pull request:
The find_package() calls of Oracle and SQLite3 emitted warnings because
their respective _FOUND variables were only set in their uppercase
variants. Letting find_package_handle_standard_args() handle the mixed
case variable and setting the uppercase variable manually gets rid of
those warnings.

Fixes #932

vadz commented on cmake/modules/FindPostgreSQL.cmake:
```diff
@@ -168,7 +168,7 @@ include(FindPackageHandleStandardArgs)
                                   REQUIRED_VARS POSTGRESQL_LIBRARY POSTGRESQL_INCLUDE_DIR
                                   VERSION_VAR POSTGRESQL_VERSION_STRING)
-set(POSTGRESQL_FOUND  ${POSTGRESQL_FOUND})
+set(POSTGRESQL_FOUND  ${PostgreSQL_FOUND})
```
Same question as above (and also below).

vadz reviewed (commented):
Thanks, I'm glad to see that this works, but I think it would be tidier to rename the variables rather keeping both versions.

Or is there some mysterious CMake lore that would make this a bad idea?

zann1x commented on cmake/modules/FindOracle.cmake:
```diff
@@ -81,7 +81,8 @@ set(ORACLE_LIBRARIES ${ORACLE_LIBRARY})
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(ORACLE DEFAULT_MSG ORACLE_LIBRARY ORACLE_INCLUDE_DIR)
+find_package_handle_standard_args(Oracle DEFAULT_MSG ORACLE_LIBRARY ORACLE_INCLUDE_DIR)
+set(ORACLE_FOUND ${Oracle_FOUND})
```
CMake variables are case-sensitive, so I assume that is why SOCI's CMake files have checks for both lower and uppercase names all over the place. I just wasn't sure if I would break something along the chain by changing this.

But if it's only used internally, I'd be down for switching to `Oracle_FOUND` only as well. This would also comply to the CMake standard: https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#standard-variable-names, which would mean that we should probably also use `Oracle_INCLUDE_DIR`, `Oracle_LIBRARY` etc. instead.
I'll look into it on the weekend.

zann1x commented:
TIL that find_package_handle_standard_args() actually already sets Oracle_FOUND as well as ORACLE_FOUND for backwards compatibility reasons.
I updated the general naming of the backend variables nonetheless to adhere to the CMake standard.

zann1x commented:
Hmm... as I said, I renamed the variables according to their package names. That makes me wonder whether we should keep the _INCLUDE_DIR and the _LIBRARY variable naming as before because those can be set by the user (as seen in the failing test) and only rename the resulting variables. Or we just keep the old state and drop the commit where I introduced all those changes.

vadz commented:
I think it would be better to avoid breaking the existing user build scripts/systems, i.e. continuing to honour the upper-case versions if they're set. But the resulting variables can be renamed, AFAICS nothing outside of SOCI should ever depend on them.

Of course, there is also the question of how much effort exactly should we spend on fixing these harmless warnings...

zann1x commented:
> Of course, there is also the question of how much effort exactly should we spend on fixing these harmless warnings...

Good point. I'll just revert the changes then and squash them together to a single commit that only removes the warnings.

vadz commented:
Thanks, looks good to me so merged now!
